streamlit/mmm-explainer/pages/Saturation.py

```diff
@@ -0,0 +1,385 @@
+#   Copyright 2024 The PyMC Labs Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+# Import custom functions
+import adstock_saturation_functions as asf
+import numpy as np
+import pandas as pd
+import plotly.graph_objects as go
+
+import streamlit as st
+
+# -------------------------- TOP OF PAGE INFORMATION -------------------------
+
+# Set browser / tab config
+st.set_page_config(
+    page_title="MMM App - Saturation Curves",
+    page_icon="🧊",
+)
+
+# Give some context for what the page displays
+st.title("Saturation Curves")
+st.markdown(
+    "This page demonstrates the forms and shapes of saturation curves for MMM.\
+             These curves try to model the relationship between weekly marketing \
+            spends for a given channel (holding other channels constant) and \
+             the conversions that result from that spend.\
+            \n It doesn't need to be conversions, it could be sales or customers acquired\
+             - whatever target metric you are interested in.\
+            "
+)
+
+st.markdown(
+    "**Reminder:** \n \
+- Certain saturation functions have **_:red[concave or convex shapes]_**  \n\
+- Certain saturation functions have **_:red[S-shapes]_**"
+)
+
+# -------------------------- SATURATION PLOTS -------------------------
+
+# Generate simulated marketing data
+np.random.seed(42)
+num_points = 500
+media_spending = np.linspace(0, 1000, num_points)  # x-axis
+# Scaling factor for nicer plotting
+
+
+# Generate simulated datasets with noise
+dummy_root = asf.root_saturation(media_spending, alpha=0.3) + np.random.normal(
+    0, 0.3, num_points
+)
+dummy_hill = asf.threshold_hill_saturation(
```

Comment:
A couple of recommendations here, instead of recreating all the functions that we have in the transformer module, it is better to import them directly and use `.eval()` to return the array. I think this will help centralize the work and prevent us from having duplicates that we must maintain.

Intead of:
```python
import adstock_saturation_functions as asf

dummy_logistic = asf.logistic_saturation(media_spending, lam=0.01) + np.random.normal(
    0, 0.1, num_points
)
```

Should be:
```python
from pymc_marketing.mmm.transformers import LogisticSaturation

dummy_logistic = LogisticSaturation(x=media_spending, lam=0.01).eval() + np.random.normal(
    0, 0.1, num_points
)
```

This means we can eliminate the file `adstock_saturation_function.py`. At the same time, this also constraint to only have functions availables in PyMC Marketing, and we should be able to add the new ones as they are implemented into the transformer script.

Reply:
This was great feedback, thanks @cetagostini - I believe I've addressed it 🙇